extensions/amp-form/0.1/amp-form.js

```diff
@@ -86,20 +90,24 @@ export class AmpForm {
 
   /** @private */
   installSubmitHandler_() {
-    this.form_.addEventListener('submit', e => this.handleSubmit_(e));
+    this.form_.addEventListener('submit', e => this.handleSubmit_(e), true);
   }
 
   /**
+   * Note on stopImmediatePropagation usage here, it is important to emulate native
+   * browser submit event blocking. Otherwise any other submit listeners would get the
+   * event.
+   *
+   * For example, action service shouldn't trigger 'submit' event if form is actually
+   * invalid. stopImmediatePropagation allows us to make sure we don't trigger it
```

Comment:
Not entirely sure if I understand but even with capture, how would you stop the event from propagating to the listener in `actions-impl`? 

It seems for `tap` event we're using `e.defaultPrevented` to prevent triggering in actions. But we can't use this with `submit` event because we always `preventDefault` when we do `xhr` submission, so we can't use that signal to prevent action from triggering.